microsoft/testsuites/vm_extensions/azure_keyvault_vm_extension.py

```diff
@@ -0,0 +1,231 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import random
+import time
+from typing import List
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.base_tools.service import Service
+from lisa.operating_system import BSD
+from lisa.sut_orchestrator.azure.common import (
+    add_system_assign_identity,
+    assign_access_policy_to_vm,
+    check_certificate_existence,
+    create_certificate,
+    create_keyvault,
+    delete_certificate,
+    delete_keyvault,
+    get_node_context,
+    get_object_id,
+    get_tenant_id,
+    rotate_certificates,
+)
+from lisa.sut_orchestrator.azure.features import AzureExtension
+from lisa.sut_orchestrator.azure.platform_ import AzurePlatform, AzurePlatformSchema
+from lisa.testsuite import TestResult
+from lisa.tools.ls import Ls
+from lisa.util import LisaException
+
+
+def _check_system_status(node: Node, log: Logger) -> None:
+    # Check the status of the akvvm_service service using the Service tool
+    service = node.tools[Service]
+    if service.is_service_running("akvvm_service.service"):
+        log.info("akvvm_service is running")
+    else:
+        log.error("akvvm_service is not running")
+        raise LisaException("akvvm_service is not running. Test case failed.")
+
+    # List the contents of the directory
+    ls = node.tools[Ls]
+    directory_contents = ls.run(
+        "/var/lib/waagent/Microsoft.Azure.KeyVault -la", sudo=True
+    ).stdout
+    log.info(f"Directory contents: {directory_contents}")
+
+
+@TestSuiteMetadata(
+    area="vm_extension",
+    category="functional",
+    description="BVT for Azure Key Vault Extension",
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class AzureKeyVaultExtensionBvt(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        The following test case validates the Azure Key Vault Linux
+        * Extension while creating the following resources:
+        * A Key Vault
+        * Two certificates in the Key Vault
+        * Retrieval of the certificate's secrets
+        through SecretClient class from Azure SDK.
+        * Installation of the Azure Key Vault Linux Extension on the VM.
+        * Installation of the certs through AKV extension
+        * Rotation of the certificates
+        * Printing the cert after rotation from the VM
+        * Deletion of the resources
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            supported_features=[AzureExtension], unsupported_os=[BSD]
+        ),
+    )
+    def verify_key_vault_extension(
+        self, log: Logger, node: Node, result: TestResult
+    ) -> None:
+        # Section for vault name and supported OS check
+        vault_name = f"kve-{time.strftime('%y%m%d%H%M%S')}-{random.randint(1, 1000):03}"
```

Comment:
The vault name should be consistent for a subscription, so the key vault will be reused in next run. Only the certificate name should be a random one to avoid conflict on different test cases. You can refer to other place, use the last chars of subscription id to get a consistent key vault name.